compiler/rustc_interface/src/util.rs

```diff
@@ -694,16 +694,25 @@ pub fn build_output_filenames(
     }
 }
 
-// Note: Also used by librustdoc, see PR #43348. Consider moving this struct elsewhere.
-//
-// FIXME: Currently the `everybody_loops` transformation is not applied to:
-//  * `const fn`, due to issue #43636 that `loop` is not supported for const evaluation. We are
-//    waiting for miri to fix that.
-//  * `impl Trait`, due to issue #43869 that functions returning impl Trait cannot be diverging.
-//    Solving this may require `!` to implement every trait, which relies on the an even more
-//    ambitious form of the closed RFC #1637. See also [#34511].
-//
-// [#34511]: https://github.com/rust-lang/rust/issues/34511#issuecomment-322340401
+/// Replaces function bodies with `loop {}` (an infinite loop). This gets rid of
+/// all semantic errors in the body while still satisfying the return type,
+/// except in certain cases, see below for more.
+///
+/// This pass is known as `everybody_loops`. Very punny.
+///
+/// Since [#73566], rustdoc no longer runs `everybody_loops`. As of February 2021,
+/// `everybody_loops` is only run for the `-Z unpretty=everybody_loops` CLI option.
```

Comment:
Does this need to be mentioned? This seems only of historical interest, and you can find it easily with git or a google search.

Reply:
> This seems only of historical interest, and you can find it easily with git or a google search.

I wanted to make it clear that it's only used for debugging. Someone reading the code might not think of or might not want to spend the time of doing a search.

Reply:
Ok, then I would say something like "This is used for the `-Z unpretty=everybody_loops` debugging pass"`. I don't think you need to mention rustdoc.

Reply:
Personally, I would still prefer to mention rustdoc, but it doesn't matter that much so I removed the sentence about rustdoc :)